src/main/java/in/projecteka/consentmanager/user/UserService.java

```diff
@@ -130,7 +130,8 @@
         if (!validateOtpVerification(otpVerification)) {
             throw new InvalidRequestException("invalid.request.body");
         }
-        return signupService.getUserName(otpVerification.getSessionId())
+        String sessionIdWithAction = SendOtpAction.RECOVER_PASSWORD.toString() + otpVerification.getSessionId();
```

Comment:
It's not a good idea to hardcode the SendOtpAction here because this function is called in multiple flows.

Reply:
But the OtpAttempt Action is also OTP_SUBMIT_RECOVER_PASSWORD and the name of the function is also verifyOtpForForgetPassword so, shouldn't it just for RECOVER_PASSWORD?

Reply:
Oh,  I thought this is in `validateAndVerifyOtp` method. Because it shows like that in git diff.